### PR TITLE
feat: Add some of the missing pieces in the create product form

### PR DESCRIPTION
**Pull request by sradevski**
The changes are still partial, there is more work to be done to have everything function properly.
Also, refactored the file upload from the product media form to a separate component

### CODE REVIEW

**Comment by kasperkristensen** [comment]
@sradevski just something to discuss: I have omitted placeholders on purpose for the following reasons:

1. They provide little to no value.
2. The increase the number of translations keys by a lot, which increases the workload of people looking to translate the UI for very little benefit.

So I would prefer to skip them entirely, except for the very few cases where they make sense, eg. search fields, native selects where we need a "dummy" placeholder value, etc. But I am open to including them if we think they provide something useful for the users.

File: packages/admin-next/dashboard/public/locales/en-US/translation.json [hunk 1]
```diff
@@ -151,7 +152,7 @@
     "attributes": "Attributes",
     "editProduct": "Edit Product",
     "editAttributes": "Edit Attributes",
-    "organization": "Organization",
+    "organization": "Organize",
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/public/locales/en-US/translation.json:155]: nit(non-blocking): update the key to match

**Review by kasperkristensen — APPROVED** [review]
LGTM, only thing is the discussion on placeholders, that I think we need to align on 👍

**Comment by sradevski** [comment]
> @sradevski just something to discuss: I have omitted placeholders on purpose for the following reasons:
> 
> 1. They provide little to no value.
> 2. The increase the number of translations keys by a lot, which increases the workload of people looking to translate the UI for very little benefit.
> 
> So I would prefer to skip them entirely, except for the very few cases where they make sense, eg. search fields, native selects where we need a "dummy" placeholder value, etc. But I am open to including them if we think they provide something useful for the users.

@kasperkristensen got it on the placeholders, I do agree that other than the UI looking a bit "more full" they don't provide a lot of value. I simply followed the designs, but I'm happy to remove them if we are happy with it design-wise 👍

**Comment by kasperkristensen** [comment]
> > @sradevski just something to discuss: I have omitted placeholders on purpose for the following reasons:
> > 
> > 1. They provide little to no value.
> > 2. The increase the number of translations keys by a lot, which increases the workload of people looking to translate the UI for very little benefit.
> > 
> > So I would prefer to skip them entirely, except for the very few cases where they make sense, eg. search fields, native selects where we need a "dummy" placeholder value, etc. But I am open to including them if we think they provide something useful for the users.
> 
> @kasperkristensen got it on the placeholders, I do agree that other than the UI looking a bit "more full" they don't provide a lot of value. I simply followed the designs, but I'm happy to remove them if we are happy with it design-wise 👍

Alright if you agree, I think we should remove them, and Ludvig hasn't commented on it when he's reviewed what has been done so far, so don't think he will mind 😅

**Pull request merged**